### PR TITLE
do not add validation context for SIMPLE

### PR DESCRIPTION
**Pull request by ramaraochavali**
We are currently adding validation context for SIMPLE mode which is not needed if cacerts are not present and is failing if gateway specifies certs via files without caCertificate, as `model.GetRootResourceName` returns `ROOTCA` as cert is not provided and SDS fails to fetch.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@howardjohn can you PTAL?

**Comment by howardjohn** [comment]
> We are currently adding validation context for SIMPLE mode which is not needed 

This is needed? We still want to verify the cert, otherwise TLS can be MITM'd? Maybe I misunderstood

File: pilot/pkg/networking/core/v1alpha3/gateway.go [hunk 1]
```diff
@@ -483,29 +483,31 @@ func buildGatewayListenerTLSContext(
...
 		}
-		authn_model.ApplyToCommonTLSContext(ctx.CommonTlsContext, certProxy, server.Tls.SubjectAltNames, []string{})
-	}
-
-	ctx.RequireClientCertificate = proto.BoolFalse
-	if server.Tls.Mode == networking.ServerTLSSettings_MUTUAL ||
-		server.Tls.Mode == networking.ServerTLSSettings_ISTIO_MUTUAL {
-		ctx.RequireClientCertificate = proto.BoolTrue
+		if server.Tls.Mode == networking.ServerTLSSettings_MUTUAL {
+			certProxy.Metadata.TLSServerRootCert = server.Tls.CaCertificates
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/gateway.go:508]: you can still set CaCertificates on SIMPLE mode though? to verify the CA?

File: pilot/pkg/networking/core/v1alpha3/gateway_test.go [hunk 2]
```diff
@@ -264,31 +237,6 @@ func TestBuildGatewayListenerTlsContext(t *testing.T) {
 							},
 						},
 					},
-					ValidationContextType: &auth.CommonTlsContext_CombinedValidationContext{
-						CombinedValidationContext: &auth.CommonTlsContext_CombinedCertificateValidationContext{
-							DefaultValidationContext: &auth.CertificateValidationContext{},
-							ValidationContextSdsSecretConfig: &auth.SdsSecretConfig{
-								Name: "file-root:ca-cert.crt",
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/gateway_test.go:271]: this looks wrong to me

File: pilot/pkg/networking/core/v1alpha3/gateway.go (hunk 1 above)
Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/gateway.go:508]: Yeah, I misunderstood the required I guess. Let me change it to add validation if CA cert is present,.

**Comment by ramaraochavali** [comment]
@howardjohn Fixed. PTAL.

**Review by howardjohn — COMMENTED** [review]
Sorry I realized my original comments maybe made no sense - I was thinking of client side (DR) not Gateway...

Why do we ever add validation context for SIMPLE? seems odd.. I guess it just means the mtls is optional though?

**Review by howardjohn — APPROVED** [review]

File: pilot/pkg/networking/core/v1alpha3/gateway.go (hunk 1 above)
Comment (howardjohn) [reply pilot/pkg/networking/core/v1alpha3/gateway.go]: Sorry I was thinking this was for dest rule not gateway. I don't know what we should do here, lets get some extra feedback

**Review by howardjohn — COMMENTED** [review]
Ok, sorry for the churn. I was confused the first time i looked at it. I spoke with @costinm and it sounds like the original change - never setting validation context for SIMPLE - is correct. If we are not setting `require_client_certificate` then we will not even request a client cert in the TLS handshake so having the validation context seems pretty pointless?

**Comment by ramaraochavali** [comment]
> If we are not setting require_client_certificate then we will not even request a client cert in the TLS handshake so having the validation context seems pretty pointless?

That was my original thought. Let me revert to that state.

**Comment by ramaraochavali** [comment]
@howardjohn I changed it to not set validation context for SIMPLE. PTAL.

**Review by howardjohn — APPROVED** [review]
thanks, sorry about the confusion

**Pull request merged**